Vector/ViewController/RoomParticipantsViewController.m

```diff
@@ -515,40 +442,78 @@ - (void)removeParticipantByKey:(NSString*)key
 
 - (void)finalizeParticipantsList
 {
-    // Sort contacts in alphabetical order (Use sortingDisplayName in which symbols are skipped)
-    NSComparator comparator = ^NSComparisonResult(Contact *contact1, Contact *contact2) {
-        
-        if (contact1.sortingDisplayName.length && contact2.sortingDisplayName.length)
+    // Sort contacts by last active, with "active now" first.
+    // ...and then by power
+    // ...and then alphabetically.
+    // We could tiebreak instead by "last recently spoken in this room" if we wanted to.
+    NSComparator comparator = ^NSComparisonResult(Contact *contactA, Contact *contactB) {
+
+        MXUser *userA = [self.mxRoom.mxSession userWithUserId:contactA.mxMember.userId];
+        MXUser *userB = [self.mxRoom.mxSession userWithUserId:contactB.mxMember.userId];
+
+        if (!userA && !userB)
         {
-            return [contact1.sortingDisplayName compare:contact2.sortingDisplayName options:NSCaseInsensitiveSearch];
+            return NSOrderedSame;
```

Comment:
Why don't you use contactA.sortingDisplayName and contactB.sortingDisplayName to order them?